### PR TITLE
Adding debug APIs using pprof

### PR DESCRIPTION
**Pull request by Avish34**
Changes:
- Added a REST server that serves debug routes.
- Created a new package for having all the constants.

Testing:
Tested the pprof using rest server and was able to visualise with Graphviz.

Sample command:
```
─╯ go tool pprof http://localhost:8081/debug/pprof/goroutine      <<< command to get goroutines

                                                 
  Fetching profile over HTTP from http://localhost:9001/debug/pprof/goroutine
  Saved profile in /Users/avish/pprof/pprof.propeller.goroutine.003.pb.gz
  File: propeller
  Type: goroutine
  Time: Dec 22, 2024 at 10:23am (IST)
  Entering interactive mode (type "help" for commands, "o" for options)
  (pprof) 
```

### CODE REVIEW

**Comment by Avish34** [comment]
@Abhishekvrshny please review.

File: cmd/service/main.go [hunk 1]
```diff
@@ -83,6 +86,16 @@ func Run(ctx context.Context, component component.IComponent) error {
 		done <- true
 	}()
 
+	url := fmt.Sprintf(":%v", global.DebugServerPort)
```
Comment (Abhishekvrshny) [inline cmd/service/main.go:89]: A HTTP server is already initialised in propeller for prometheus metrics and webscokets. Possible to re-use the same server? It's initialised here: https://github.com/CRED-CLUB/propeller/blob/main/internal/component/api_server_component.go

File: internal/debug/debug.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+// Stop the rest server
+func (ds *DebugServer) Stop(pctx context.Context) {
+	ctx, cancel := context.WithTimeout(pctx, 5*time.Second)
+	defer cancel()
+	ds.server.Shutdown(ctx)
+}
+
+// registerDebugRoutes registers all the debug routes
+func (ds *DebugServer) registerDebugRoutes() {
```
Comment (Abhishekvrshny) [inline internal/debug/debug.go:51]: is it possible to put registering these debug routes behind a config to enable profiling? The config file is defined at: https://github.com/CRED-CLUB/propeller/blob/main/config/propeller.toml

**Review by Abhishekvrshny — CHANGES_REQUESTED** [review]

File: cmd/service/main.go (hunk 1 above)
Comment (Avish34) [reply cmd/service/main.go]: done

File: internal/debug/debug.go (hunk 2 above)
Comment (Avish34) [reply internal/debug/debug.go]: These routers are static. I feel the config file should be used to drive user provided parameters. I have pushed new changes that won't require registering routes now.

**Comment by Abhishekvrshny** [comment]
@Avish34  it is always safe to put profiling related endpoints behind a config flag as you do not want someone to inadvertently run profiling in prod as it may have performance implications. The same approach is handled by other open-source projects, like how `kubelet` config has a parameter called `enableProfilingHandler` to explicitly enable or disable pprof handlers. Ref: https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/

File: internal/component/api_server_component.go [hunk 3]
```diff
@@ -118,6 +118,8 @@ func (web *APIServer) Start(ctx context.Context) error {
 			ws.WebSocketConnect,
 		)
 
+		// debug handlers for profiling
+		m.Handle("/debug/", http.DefaultServeMux)
```
Comment (Abhishekvrshny) [inline internal/component/api_server_component.go:122]: is `import _ "net/http/pprof"` not required?

**Comment by Avish34** [comment]
@Abhishekvrshny makes sense. Added the config parameter to the config file.

File: config/propeller.toml [hunk 4]
```diff
@@ -5,6 +5,7 @@ ClientHeader = "x-user-id"
 EnableDeviceSupport = true
 DeviceHeader = "x-device-id"
 DeviceAttributeHeaders = ["x-os", "x-os-version"]
+EnableDebuggingHandlers = true
```
Comment (Abhishekvrshny) [inline config/propeller.toml:8]: can we call this `EnableProfilingHandlers` which is a more specific and apt term here?

Comment (Avish34) [reply config/propeller.toml]: done

**Review by Abhishekvrshny — APPROVED** [review]

**Comment by Abhishekvrshny** [comment]
Thanks for the contribution @Avish34 !

**Pull request merged**